libp2p/autotls/acme/mockapi.nim

```diff
@@ -0,0 +1,36 @@
+import chronos, chronos/apps/http/httpclient, json
+
+import ./api, ./utils
+
+export api
+
+type MockACMEApi* = ref object of ACMEApi
+  mockedHeaders*: HttpTable
+  mockedBody*: JsonNode
+
+proc new*(
+    T: typedesc[MockACMEApi], acmeServerURL: string = LetsEncryptURL
```

Comment:
is for mock `acmeServerURL` really needed?

Reply:
that usecase will probably never happen. and is probably not ideal approach.

what happens when someone adds new property to constructor to non-mocked struct? do they have to add property to mocked struct? will they remember to do that? what if they forgot, then why have rule if it is not enforced?
what happens when someone adds new property to mock, lets say mock has to be created with something additional to regular field. do you add same field to non-mock struct?

it's always better to remove anything unused. or if you don't do it now, other developer will clean it up after you.

if you want to have drop in replacment then it's solved via different approach.